Story/x1175 (#612)
unipartdigital/udes-open: addons/udes_stock/README.md

| Helpers                         | Description                                                                                                                                                                                                                                                                                                                                       |
|---------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| get_empty_locations             | Returns the recordset of locations that are child of the instance dest location and are empty.Expects a singleton instance.                                                                                                                                                                                                                       |

> | _get_child_dest_locations       | Return the child locations of the instance dest location. Extra domains are added to the child locations search query, when specified. Expects a singleton instance.                                                                                                                                                                              |

Reviewer: ```suggestion
| _get_child_dest_locations       | Returns the child locations of the instance dest location. Extra domains are added to the child locations search query, when specified. Expects a singleton instance.                                                                                                                                                                              |
```

> | get_move_lines                  | Get move lines associated to picking, uses functions in stock_move_line.                                                                                                                                                                                                                                                                          |

Reviewer: ```suggestion
| get_move_lines                  | Gets move lines associated to picking, uses functions in stock_move_line.                                                                                                                                                                                                                                                                          |
```

| backorder_move_lines            | Checks if a backorder is required, creates one according to _backorder_move_lines if so, else returns an empty record set; neither picking is validated.                                                                                                                                                                                          |
| _backorder_move_lines           | Creates a backorder pick from self for all incomplete work, the original move retains with all completed work, new moves are created for the incomplete wrok.  The original pick is not validated. It should be called from within backorder_move_lines.                                                                                          |
| _requires_backorder             | Checks if a backorder is required by checking if all move lines within a picking are present in mls. Cannot be consolidated with _check_backorder in Odoo core, because it does not take into account any move lines parameter.                                                                                                                   |

> | _prepare_picking_info           | Prepare the picking_info and products_info, returning the picking_info as a list of pickings, and products_info None or as a list of lists of a dictionary of the product info. This is to allow us to create multiple pickings at once.                                                                                                          |

Reviewer: ```suggestion
| _prepare_picking_info           | Prepares the picking_info and products_info, returning the picking_info as a list of pickings, and products_info None or as a list of lists of a dictionary of the product info. This is to allow us to create multiple pickings at once.                                                                                                          |
```

> | _prepare_move                   | Return a list of the move details to be used later in creation of the move(s). Note: pickings and products_info should be formatted via _prepare_picking_info. Allows for multiple moves to be created at once.                                                                                                                                   | 

Reviewer: ```suggestion
| _prepare_move                   | Returns a list of the move details to be used later in creation of the move(s). Note: pickings and products_info should be formatted via _prepare_picking_info. Allows for multiple moves to be created at once.                                                                                                                                   | 
```

| _create_batch                   | Creates a batch                                                                                                                                                                                                                                                                                                                                   |

> | create_picking                  | Create and return a picking for the given picking_type. For multiple pickings a list of lists of dicts of product_info should be passed, and pickings with the same picking_type and other kwargs are the same. The kwargs are applied to pickings, not moves. If needed, the moves can be created outside of create_pickings with _create_moves. |

Reviewer: ```suggestion
| create_picking                  | Creates and returns a picking for the given picking_type. For multiple pickings a list of lists of dicts of product_info should be passed, and pickings with the same picking_type and other kwargs are the same. The kwargs are applied to pickings, not moves. If needed, the moves can be created outside of create_pickings with _create_moves. |
```

> | _create_move                    | Create and return move(s) for the given move_values. Should be used in conjunction with _prepare_move to obtain move_values                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| _create_move                    | Creates and returns move(s) for the given move_values. Should be used in conjunction with _prepare_move to obtain move_values                                                                                                                                                                                                                       |
```
